src/servers/src/http/authorize.rs

```diff
@@ -281,8 +258,33 @@ fn need_auth<B>(req: &Request<B>) -> bool {
     path.starts_with(HTTP_API_PREFIX)
 }
 
+fn extract_db_from_query(query: &str) -> Option<&str> {
+    for pair in query.split('&') {
+        if let Some(db) = pair.strip_prefix("db=") {
+            return if pair.len() == 3 { None } else { Some(db) };
+        }
+    }
+    None
+}
+
+fn extract_influxdb_user_from_query(query: &str) -> (Option<&str>, Option<&str>) {
+    let mut username = None;
+    let mut password = None;
+
+    for pair in query.split('&') {
+        if pair.starts_with("u=") && pair.len() > 2 {
+            username = Some(&pair[2..]);
```

Comment:
Can we use `strip_prefix` like how we parse `db`?

Reply:
```Rust
        if let Some(u) = pair.strip_prefix("u=") {
            username = if u.is_empty() { None } else { Some(u) };
        } else if let Some(p) = pair.strip_prefix("p=") {
            password = if p.is_empty() { None } else { Some(p) };
        }